### PR TITLE
Allow discount cache to be reset

### PR DESCRIPTION
**Pull request by ryanmitchell**
This PR adds the ability to reset the DiscountManager query cache - this is necessary if you add a discount coupon and want to update the cart totals in the same request.

### CODE REVIEW

File: packages/core/src/Managers/DiscountManager.php [hunk 1]
```diff
@@ -209,6 +209,13 @@ public function apply(Cart $cart): Cart
         return $cart;
     }
 
+    public function resetDiscounts(): self
```
Comment (alecritson) [inline packages/core/src/Managers/DiscountManager.php:215]: Do you think `clearDiscounts` would be better? 🤔 @glennjacobs

Comment (glennjacobs) [reply packages/core/src/Managers/DiscountManager.php:215]: No, I think reset is correct.

**Comment by glennjacobs** [comment]
@ryanmitchell I know the coupon stuff has tripped up devs a fair amount historically. Would it make sense for us to put this into the docs also?

**Comment by ryanmitchell** [comment]
Yeah I think it would. I’ll add that to this PR at some point tomorrow

**Comment by glennjacobs** [comment]
> Yeah I think it would. I’ll add that to this PR at some point tomorrow

I presume this solves the coupon issue, so we could do with a test to demonstrate that also pretty pls :-)

**Comment by ryanmitchell** [comment]
@glennjacobs docs added. I tried to write a test but it's doing the same as `can_get_discount_with_coupon`, but that calls getDiscounts directly rather than bypassing the cache. If theres another test I can piggy-back on do let me know.

**Comment by glennjacobs** [comment]
> @glennjacobs docs added. I tried to write a test but it's doing the same as `can_get_discount_with_coupon`, but that calls getDiscounts directly rather than bypassing the cache. If theres another test I can piggy-back on do let me know.

Thanks, I'll take a look.

**Review by glennjacobs — APPROVED** [review]

**Pull request merged**